detection_rules/rule.py

```diff
@@ -974,10 +991,22 @@ def latest_version(self) -> Optional[int]:
     def autobumped_version(self) -> Optional[int]:
         """Retrieve the current version of the rule, accounting for automatic increments."""
         version = self.latest_version
+        toml_version = self.data.get("version")
+
+        if toml_version:
+            if BYPASS_VERSION_LOCK:
+                version = toml_version
+            else:
+                print(f"WARNING: Rule {self.name} - {self.id} has a version set in the rule TOML."
```

Comment:
I combined a few of these comments in https://github.com/elastic/detection-rules/pull/3657/commits/252b1fe46cedb723f54fda2e7ffc6fb771a61007 

It's meant to convey the conceptual differences so we can all see what it would look like.

- I don't think we need to reconcile the `self.autobumped_version` calls because it already loads the version from `self.latest_version`
- I think leaving the version to 1 makes sense. 
- This commit basically doesn't load the lock files at all. 


One thing we can probably do at the end is standardize the warnings.

Also do we want to add bypass checks in the `from_dict` / `to_dict` methods? It will at least provide some coverage for if users ever call methods directly.

Reply:
@brokensound77 @traut @eric-forte-elastic I've update the summary with several commands used to test and their outputs. If there's another specific command or unit test I should include, lmk.